homeassistant/components/climate/xiaomi_miio.py

```diff
@@ -0,0 +1,624 @@
+"""
+Support SmartMi acpartner.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/climate.xiaomi_miio
+"""
+import logging
+import asyncio
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.components.climate import (
+    PLATFORM_SCHEMA,
+    ClimateDevice, ATTR_TARGET_TEMP_HIGH, ATTR_TARGET_TEMP_LOW)
+from homeassistant.const import (
+    TEMP_CELSIUS, ATTR_TEMPERATURE, ATTR_UNIT_OF_MEASUREMENT,
+    CONF_NAME, CONF_HOST, CONF_TOKEN, CONF_TIMEOUT)
+# from homeassistant.helpers import condition
+from homeassistant.helpers.event import (
+    async_track_state_change, async_track_time_interval)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['python-miio==0.3.2']
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['sensor']
+
+DEFAULT_TOLERANCE = 0.3
+DEFAULT_NAME = 'Mi ACpartner'
+
+DEFAULT_TIMEOUT = 10
+DEFAULT_RETRY = 3
+
+DEFAULT_MIN_TMEP = 16
+DEFAULT_MAX_TMEP = 30
+DEFAULT_STEP = 1
+
+CONF_SENSOR = 'target_sensor'
+# CONF_TARGET_TEMP = 'target_temp'
+CONF_SYNC = 'sync'
+CONF_CUSTOMIZE = 'customize'
+
+__Presets__ = {
```

Comment:
This looks a lot like protocol-level magic strings. We try to keep protocol implementations out of home assistant as far as possible. Perhaps it would be better to include this low-level protocol stuff in the `python-miio` package?